### PR TITLE
Feature/r73

### PR DESCRIPTION
**Pull request by fblundun**

### CODE REVIEW

File: 3-enrich/scala-common-enrich/src/main/scala/com.snowplowanalytics.snowplow.enrich/common/utils/ScalazArgs.scala [hunk 1]
```diff
@@ -13,36 +13,39 @@
...
 import iglu.client.validation.ProcessingMessageMethods._
 
-// TODO: move this into scala-util
-
 /**
  * Module to hold the pimped class
  */
 object ScalazArgs {
 
+  type ListableArg = {
```
Comment (alexanderdean) [inline 3-enrich/scala-common-enrich/src/main/scala/com.snowplowanalytics.snowplow.enrich/common/utils/ScalazArgs.scala]: Can we add a docstring explaining why we are using the structural type here?

File: 3-enrich/scala-hadoop-shred/src/main/scala/com.snowplowanalytics.snowplow.enrich/hadoop/ShredJob.scala [hunk 2]
```diff
@@ -108,6 +108,49 @@ object ShredJob {
...
   import Dsl._
   val ShreddedPartition = new UrShreddedPartition('schema)
+
+  // Indexes for the contexts, unstruct_event, and derived_contexts fields
+  private val IgnoredJsonFields = Set(52, 58, 122)
+
+  private val alteredEnrichedEventSubdirectory = "atomic-events"
+
+  /**
+   * Ready the enriched event for database load by removing JSON fields
```
Comment (alexanderdean) [inline 3-enrich/scala-hadoop-shred/src/main/scala/com.snowplowanalytics.snowplow.enrich/hadoop/ShredJob.scala]: " and truncating field lengths based on Postgres' column types."

File: 4-storage/postgres-storage/sql/atomic-def.sql [hunk 3]
```diff
@@ -9,7 +9,7 @@
 -- "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
 -- See the Apache License Version 2.0 for the specific language governing permissions and limitations there under.
 --
--- Version:     0.6.0
+-- Version:     0.7.0
```
Comment (alexanderdean) [inline 4-storage/postgres-storage/sql/atomic-def.sql]: We're already on 0.7.0 - this is 0.8.0?

Comment (alexanderdean) [inline 4-storage/postgres-storage/sql/atomic-def.sql]: Gah ignore me, this is Postgres.

File: 3-enrich/scala-hadoop-shred/src/main/scala/com.snowplowanalytics.snowplow.enrich/hadoop/ShredJob.scala [hunk 4]
```diff
@@ -108,6 +108,49 @@ object ShredJob {
...
+   *
+   * @param enrichedEvent TSV
+   * @return the same TSV with the JSON fields removed
+   */
+  def alterEnrichedEvent(enrichedEvent: String): String = {
+    import common.utils.ConversionUtils
+
+    (enrichedEvent.split("\t", -1).toList, PostgresConstraints.maxFieldLengths)
```
Comment (alexanderdean) [inline 3-enrich/scala-hadoop-shred/src/main/scala/com.snowplowanalytics.snowplow.enrich/hadoop/ShredJob.scala]: Add a comment here:

```scala
// TODO: move PostgresConstraints code out into Postgres-specific shredder when ready.
// It's okay to apply Postgres constraints to events being loaded into Redshift as the PG
// constraints are typically more permissive, but Redshift will be covered by the
// COPY ... TRUNCATECOLUMNS.
```

**Comment by alexanderdean** [comment]
Looks great - feedback is just a few new comments to add...

**Comment by fblundun** [comment]
Ok - I've added those comments.

**Comment by alexanderdean** [comment]
Ace - thanks

**Comment by fblundun** [comment]
I really don't understand why Travis failed the DecodeBase64UrlSpec spec - it works for locally.

https://travis-ci.org/snowplow/snowplow/jobs/86191378

**Comment by alexanderdean** [comment]
Buggy openjdk?

**Comment by fblundun** [comment]
That sounds plausible...

**Comment by fblundun** [comment]
Note that I haven't yet rebased this to 1 commit per ticket

File: 4-storage/storage-loader/lib/snowplow-storage-loader/redshift_loader.rb [hunk 5]
```diff
@@ -26,6 +26,12 @@ module RedshiftLoader
       # Constants for the load process
       EVENT_FIELD_SEPARATOR = "\\t"
 
+      # Used to find the altered enriched events
+      ALTERED_ENRICHED_PATTERN = /(run=[0-9\-]+\/atomic-events)/
+
+      # Versions 0.5.0 and earlier of Hadoop Shred don't copy atomic.events into the shredded bucket
+      OLD_ENRICHED_PATTERN = /0\.[0-5]\.0/
```
Comment (alexanderdean) [inline 4-storage/storage-loader/lib/snowplow-storage-loader/redshift_loader.rb]: @fblundun - this regexp doesn't successfully capture Hadoop Shred 0.2.1

Comment (fblundun) [inline 4-storage/storage-loader/lib/snowplow-storage-loader/redshift_loader.rb]: Sorry @alexanderdean - this is fixed now with `/0\.[0-5]\.[0-9]/`.

**Pull request closed without merging**